common-libs/common/src/services/DidDocumentService/elem-lib/op.ts

```diff
@@ -18,20 +19,20 @@ export const getDidDocumentModel = (primaryPublicKey: string, recoveryPublicKey:
   ],
 })
 
-export const getCreatePayload = (didDocumentModel: any, primaryKey: any) => {
+export const getCreatePayload = (didDocumentModel: any, signFn: (payload: Buffer) => Buffer) => {
   // Create the encoded protected header.
   const header = {
     operation: 'create',
     kid: '#primary',
     alg: 'ES256K',
   }
-  return makeSignedOperation(header, didDocumentModel, primaryKey.privateKey)
+  return makeSignedOperation(header, didDocumentModel, signFn)
 }
 
-export const makeSignedOperation = (header: any, payload: any, privateKey: any) => {
+export const makeSignedOperation = (header: any, payload: any, signFn: (payload: Buffer) => Buffer) => {
```

Comment:
Similarly, this method could return `{ encodedHeader, encodedPayload }`, and let its callers to call `signFn`.